### PR TITLE
Add some requested new units : J/m², gr/ft³,gr

### LINKED ISSUE #4307: Unit °dH could not be parsed to a known unit  for Item Number:AmountOfSubstance
When configuring an item of type Number:AmountOfSubstance and specifying to use unit °dH (by adding unit="°dH" in channel-linkage) openHab keeps using unit "mol" and issues a warning message to the log such as:

`2024-07-08 11:28:33.129 [WARN ] [penhab.core.library.items.NumberItem] - **Unit '°dH' could not be parsed to a known unit. Keeping old unit 'mol' for item** 'gruenbeckRawWater'.`

Also See:
[https://community.openhab.org/t/problem-with-unit-dh-for-item-of-type-number-amountofsubstance-using-mol-instead-of-dh/157079](url)

## Expected Behavior
The system should use the specified unit °dH as per current documentation in chapter “SI (base unit symbols)”
of [https://www.openhab.org/docs/concepts/units-of-measurement.html](url)

## Current Behavior
openHab issues a warning message and keeps using the default(?) unit mol

## Possible Solution
no clue since I'm by far to novice of a developer to say ;-)

## Steps to Reproduce (for Bugs)
setup an item such as:
`Number:AmountOfSubstance     gruenbeckRawWater            "Wasserhärte Rohwasser [%.2f %unit%]"                                                        {channel="mqtt:topic:MQTTMosquittoBridge:gruenbeckSoftliqSd21:rawWater", unit="°dH"}`
and link it to any channel providing a pure number value

## Context
Encountered the issue while creating a widget for my water softener 
for context details: see [https://community.openhab.org/t/gruenbeck-water-softener-widget/157104](url)

## Your Environment
openHab 4.2.0 release build
Raspberry pi 5b 8GB
Debian GNU/Linux 12 (bookworm)
OpenJDK 64-Bit Server VM (build 17.0.11+9-Debian-1deb12u1, mixed mode

### LINKED ISSUE #4346: Add system channel for `Radiant Exposure` / `energy per unit area`
Can you please add system channel for [Radiant Exposure](https://en.wikipedia.org/wiki/Radiant_exposure)? The default unit can be 

$\frac{J}{\text{m}^2} or \frac{W \cdot s}{\text{m}^2} or \frac{kWh}{\text{m}^2}$

The main difference from intensity ($\frac{W}{\text{m}^2}$) is that radiant exposure contains time in it.

### PR DESCRIPTION
**Pull request by clinique**
Partially solves #4294 and #4143
Resolves #4307
Resolves #4346

### CODE REVIEW

File: bundles/org.openhab.core/src/main/java/org/openhab/core/library/dimension/EnergyDensity.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.library.dimension;
+
+import javax.measure.Quantity;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link EnergyDensity} defines the dimension for prices
```
Comment (jlaur) [inline bundles/org.openhab.core/src/main/java/org/openhab/core/library/dimension/EnergyDensity.java:20]:
```suggestion
 * The {@link EnergyDensity} defines the dimension for energy density
```

[the remark this answers is not in the document]
Comment (jlaur) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/library/dimension/EnergyDensity.java:20]: Reading that, is this dimension correct for e.g. Wh/m²? That seems to be something like Wh/L, i.e. per [volume](https://www.cevalogistics.com/en/glossary/energy-density):
> Energy density refers to the quantity of energy in a material per unit of volume.

whereas Wh/m² is more like unit of energy in relation to surface area?

I don't know the answers yet, only the questions. 😄

**Comment by lsiepel** [comment]
@clinique would https://github.com/openhab/openhab-addons/pull/17239#discussion_r1714219060 that also be possible to add?

File: bundles/org.openhab.core/src/main/java/org/openhab/core/library/dimension/EnergyDensity.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (lsiepel) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/library/dimension/EnergyDensity.java:20]: https://github.com/openhab/openhab-core/issues/4346

File: bundles/org.openhab.core/src/main/java/org/openhab/core/library/unit/Units.java [hunk 2]
```diff
@@ -91,13 +92,15 @@ public final class Units extends CustomUnits {
     public static final Unit<AmountOfSubstance> MOLE = addUnit(tech.units.indriya.unit.Units.MOLE);
     public static final Unit<Volume> LITRE = addUnit(tech.units.indriya.unit.Units.LITRE);
     @SuppressWarnings("unchecked")
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/library/unit/Units.java:94]: Unfortunately yes, I tried to get rid of it without suceess.

File: bundles/org.openhab.core/src/main/java/org/openhab/core/library/dimension/EnergyDensity.java (hunk 1 above)
Comment (clinique) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/library/dimension/EnergyDensity.java:20]: Done

[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/library/dimension/EnergyDensity.java:20]: Yes, you're correct; it needs to be reworded.

File: bundles/org.openhab.core/src/main/java/org/openhab/core/internal/i18n/I18nProviderImpl.java [hunk 3]
```diff
@@ -433,6 +434,7 @@ public static Map<Class<? extends Quantity<?>>, Map<SystemOfUnits, Unit<? extend
         addDefaultUnit(dimensionMap, Radioactivity.class, Units.BECQUEREL);
         addDefaultUnit(dimensionMap, SolidAngle.class, Units.STERADIAN);
         addDefaultUnit(dimensionMap, Speed.class, SIUnits.KILOMETRE_PER_HOUR, ImperialUnits.MILES_PER_HOUR);
+        addDefaultUnit(dimensionMap, SurfaceEnergy.class, Units.WATT_HOUR_PER_SQUARE_METRE);
```
Comment (jlaur) [inline bundles/org.openhab.core/src/main/java/org/openhab/core/internal/i18n/I18nProviderImpl.java:437]: Consider if default unit should be J/m²? I'm not sure what is most common, but that seems to be the SI unit.

File: bundles/org.openhab.core/src/main/java/org/openhab/core/library/dimension/EnergyDensity.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jlaur) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/library/dimension/EnergyDensity.java]: Hmm, both surface energy and radiant exposure can be expressed in unit J/m², but I'm wondering if https://en.wikipedia.org/wiki/Radiant_exposure is closer to the requested dimension in the linked issue? I guess it depends on the context. A quick ChatGTP session:
- **Radiant exposure** (denoted as 𝐻) is the energy received per unit area of a surface from radiation over a given time period.
- **Surface energy** (𝛾) is the amount of energy per unit area required to create a new surface. It quantifies the cohesive forces between molecules at a material's surface.

[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/library/dimension/EnergyDensity.java]: Yes, agreed. These notions use the same measure and we can understand these are the two sides of the same coin. You're correct that in an home automation context, we're more likely to talk about radiant exposure than surface energy (even if yesterday I did even not know these :-P)

**Comment by lsiepel** [comment]
This PR would also 'require' updates to other repo's:
- [ ] Documenation: https://github.com/openhab/openhab-docs/blob/main/concepts/units-of-measurement.md
- [ ] MainUI: https://github.com/openhab/openhab-webui/blob/main/bundles/org.openhab.ui/web/src/assets/units.js
- [ ] other ui?
- [ ] Addons: bindings currenlty using Indriya library to create the custom units need to change to core provided units.

**Comment by clinique** [comment]
@lsiepel : PR created for documentation

File: bundles/org.openhab.core/src/main/java/org/openhab/core/library/unit/Units.java [hunk 4]
```diff
@@ -91,13 +92,15 @@ public final class Units extends CustomUnits {
...
-    public static final Unit<AmountOfSubstance> DEUTSCHE_HAERTE = addUnit(new TransformedUnit<>("°dH",
-            (Unit<AmountOfSubstance>) MetricPrefix.MILLI(Units.MOLE).divide(Units.LITRE), MultiplyConverter.of(5.6)));
+    public static final Unit<Dimensionless> DEUTSCHE_HAERTE = addUnit((Unit<Dimensionless>) new TransformedUnit<>("°dH",
```
Comment (lsiepel) [inline bundles/org.openhab.core/src/main/java/org/openhab/core/library/unit/Units.java:95]: Why is this changed to Dimensionless?

**Comment by jimtng** [comment]
Suggestion: make the PR title list all the new units added. It would make for a much more informative release log

File: bundles/org.openhab.core/src/main/java/org/openhab/core/library/unit/Units.java (hunk 4 above)
Comment (clinique) [reply bundles/org.openhab.core/src/main/java/org/openhab/core/library/unit/Units.java:95]: See this [comment](https://github.com/openhab/openhab-core/issues/4307#issuecomment-2226858965)

**Comment by clinique** [comment]
> Suggestion: make the PR title list all the new units added, or at least hint at what type of units were added. It would make for a much more informative release log

Done

**Review by holgerfriedrich — COMMENTED** [review]
Overall, LGTM.
Could you please cover `JOULE_PER_SQUARE_METRE` in the tests, or did I overlook it?

**Review by holgerfriedrich — APPROVED** [review]

**Comment by holgerfriedrich** [comment]
@lsiepel should we delay this (i.e., shift to 5.0) because of further implications to other parts of OH?
https://github.com/openhab/openhab-core/pull/4467#issuecomment-2518415750

**Comment by lsiepel** [comment]
> @lsiepel should we delay this (i.e., shift to 5.0) because of further implications to other parts of OH? [#4467 (comment)](https://github.com/openhab/openhab-core/pull/4467#issuecomment-2518415750)

My first thought was to merge it so future addons would be compatible with 4.3.x core api’s but as 5.0 would be breaking anyway (Java 21) there is no point.
So I guess it is better to merge after 4.3.0 release

**Comment by clinique** [comment]
> Overall, LGTM. Could you please cover `JOULE_PER_SQUARE_METRE` in the tests, or did I overlook it?

Done

**Review by holgerfriedrich — APPROVED** [review]

**Review by lsiepel — APPROVED** [review]
In that case maybe @holgerfriedrich can merge this. Not much risk.

**Comment by holgerfriedrich** [comment]
@florian-h05 @lsiepel Can you explain what is the required change to [units.js](https://github.com/openhab/openhab-webui/blob/main/bundles/org.openhab.ui/web/src/assets/units.js) mentioned above? Will you take care?

**Comment by lsiepel** [comment]
Ping @magx2 foe salus binding
